src/fsharp/service/service.fs

```diff
@@ -3029,26 +3019,25 @@ type FSharpChecker(legacyReferenceResolver, projectCacheSize, keepAssemblyConten
                 return res
         }
 
+    member ic.MatchBraces(filename, source: string, options: FSharpProjectOptions, ?userOpName: string) =
+        let userOpName = defaultArg userOpName "Unknown"
+        let parsingOptions, _ = ic.GetParsingOptionsFromProjectOptions(options)
+        ic.MatchBraces(filename, SourceText.ofString source, parsingOptions, userOpName)
+
     member ic.GetParsingOptionsFromProjectOptions(options): FSharpParsingOptions * _ =
         let sourceFiles = List.ofArray options.SourceFiles
         let argv = List.ofArray options.OtherOptions
         ic.GetParsingOptionsFromCommandLineArgs(sourceFiles, argv, options.UseScriptResolutionRules)
 
-    member ic.MatchBraces(filename, source, options: FSharpProjectOptions, ?userOpName: string) =
-        let userOpName = defaultArg userOpName "Unknown"
-        let parsingOptions, _ = ic.GetParsingOptionsFromProjectOptions(options)
-        ic.MatchBraces(filename, source, parsingOptions, userOpName)
-
     member ic.ParseFile(filename, source, options, ?userOpName: string) =
         let userOpName = defaultArg userOpName "Unknown"
         ic.CheckMaxMemoryReached()
         backgroundCompiler.ParseFile(filename, source, options, userOpName)
 
-
-    member ic.ParseFileInProject(filename, source, options, ?userOpName: string) =
+    member ic.ParseFileInProject(filename, source: string, options, ?userOpName: string) =
```

Comment:
Is there a reason for `source` remaining?

Reply:
Deprecated API (it's marked with the ObsoleteAttribute in .fsi), it's still used in tests and other places that isn't VS. We will have a separate PR that will address the removal of these deprecated APIs.